scripts/zones/Norg/npcs/Gilgamesh.lua

```diff
@@ -49,13 +55,39 @@ end;
 -- 12 parle de kuzotz ? parle de bijoux aussi
 -- 10 parle de zitah
 function onEventUpdate(player,csid,option)
+    if ((csid == 234 ) and option == 1) then
+        player:updateEvent(15962,15963,15964,15965)
+    end
 end;
 
 function onEventFinish(player,csid,option)
 
     if (csid == 99) then
         player:tradeComplete();
         player:setCharVar("MissionStatus",3);
+    elseif (csid == 232 or csid == 234) then -- 
+        local reward = 0;
+        if (option == 1) then
+            reward = 15962; -- Static Earring
+        elseif (option == 2) then
+            reward = 15963; -- Magnetic Earring
+        elseif (option == 3) then
+            reward = 15964; -- Hollow Earring
+        elseif (option == 4) then
+            reward = 15965; -- Ethereal Earring
     end
 
-end;
+      if (reward ~= 0) then
+          if (player:getFreeSlotsCount() >= 1 and player:hasItem(reward) == false) then
+              player:addItem(reward);
+              player:messageSpecial(ID.text.ITEM_OBTAINED,reward)
+              player:completeQuest(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH)
+              player:addMission(COP, tpz.mission.id.cop.THE_LAST_VERSE)
+              player:addMission(ZILART,tpz.mission.id.zilart.THE_LAST_VERSE)
+          end
+              player:setCharVar("ApocalypseNigh",0)
+          else
+              player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED,reward)
+      end
```

Comment:
this whole function is improperly indented, and when you fix the indentation it makes some logic errors apparent.  here's what it looks like with fixed indentation:

```
function onEventFinish(player,csid,option)

    if (csid == 99) then
        player:tradeComplete();
        player:setCharVar("MissionStatus",3);
    elseif (csid == 232 or csid == 234) then -- 
        local reward = 0;
        if (option == 1) then
            reward = 15962; -- Static Earring
        elseif (option == 2) then
            reward = 15963; -- Magnetic Earring
        elseif (option == 3) then
            reward = 15964; -- Hollow Earring
        elseif (option == 4) then
            reward = 15965; -- Ethereal Earring
        end

        if (reward ~= 0) then
            if (player:getFreeSlotsCount() >= 1 and player:hasItem(reward) == false) then
                player:addItem(reward);
                player:messageSpecial(ID.text.ITEM_OBTAINED,reward)
                player:completeQuest(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH)
                player:addMission(COP, tpz.mission.id.cop.THE_LAST_VERSE)
                player:addMission(ZILART,tpz.mission.id.zilart.THE_LAST_VERSE)
            end
            player:setCharVar("ApocalypseNigh",0)
        else
            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED,reward)
        end
    end		
end;
```

this would set the player's variable to 0 when they didn't have inventory space for the reward.  it would also give an "Item cannot be obtained" message then reward was 0.